src/makeCalculator.js

```diff
@@ -4,7 +4,46 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      if (number === 0) {
+        console.log('Cannot divide by zero');
+        return;
+      }
+
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      if (typeof callback === 'function') {
+        callback.call(this, number);
```

Comment:
Using `callback.call(this, number)` is correct, but you could also use `callback.bind(this)` to bind the method to the calculator instance. This would allow you to pass the bound method directly to `operate` without needing to use `call` each time.